File: app/src/main/java/com/example/user/contactlist/data/ContactRepository.java
package com.example.user.contactlist.data;

import android.arch.lifecycle.LiveData;
import android.content.Context;
import android.database.Cursor;
import android.provider.ContactsContract;

import com.example.user.contactlist.data.local.AppDatabase;
import com.example.user.contactlist.data.model.Contact;

import java.util.List;


public class ContactRepository {

    private Context context;
    private AppDatabase database;

    public ContactRepository(Context context, AppDatabase database) {
        this.context = context;
        this.database = database;
    }

    public void saveContactsInDataBase() {

        Cursor cursor = context.getContentResolver().query(ContactsContract.CommonDataKinds.Phone.CONTENT_URI,
                null, null, null,
                ContactsContract.CommonDataKinds.Phone.DISPLAY_NAME + " ASC");

        if ((cursor != null ? cursor.getCount() : 0) > 0) {
            while (cursor.moveToNext()) {

                String name = cursor.getString(cursor.getColumnIndex(ContactsContract.Contacts.DISPLAY_NAME));
                String phoneNo = cursor.getString(cursor.getColumnIndex(ContactsContract.CommonDataKinds.Phone.NUMBER));
                String photoUri = cursor.getString(cursor.getColumnIndex(ContactsContract.CommonDataKinds.Phone.PHOTO_URI));

                Contact contact = new Contact(name, phoneNo, photoUri);
                if (database.contactDao().contains(phoneNo) == null)
                    database.contactDao().insert(contact);
            }
        }
        if (cursor != null) {
            cursor.close();
        }
    }

    public LiveData<List<Contact>> getContacts() {
        return database.contactDao().getAllContacts();
    }
}

File: app/src/main/java/com/example/user/contactlist/data/ContactRepository.kt
package com.example.user.contactlist.data

import android.arch.lifecycle.LiveData
import android.content.Context
import android.provider.ContactsContract

import com.example.user.contactlist.data.local.AppDatabase
import com.example.user.contactlist.data.model.Contact


class ContactRepository(private val context: Context, private val database: AppDatabase) {

    val contacts: LiveData<List<Contact>>
        get() = database.contactDao().allContacts

    fun saveContactsInDataBase() {

        val cursor = context.contentResolver.query(ContactsContract.CommonDataKinds.Phone.CONTENT_URI,
                null, null, null,
                ContactsContract.CommonDataKinds.Phone.DISPLAY_NAME + " ASC")

        if (cursor?.count ?: 0 > 0) {
            while (cursor!!.moveToNext()) {

                val name = cursor.getString(cursor.getColumnIndex(ContactsContract.Contacts.DISPLAY_NAME))
                val phoneNo = cursor.getString(cursor.getColumnIndex(ContactsContract.CommonDataKinds.Phone.NUMBER))
                var photoUri = cursor.getString(cursor.getColumnIndex(ContactsContract.CommonDataKinds.Phone.PHOTO_URI))
                if(photoUri == null)
                    photoUri = ""

                val contact = Contact(name, phoneNo, photoUri)
                if (database.contactDao().contains(phoneNo) == null) // fixme why is always false
                    database.contactDao().insert(contact)
            }
        }
        cursor?.close()
    }
}
